heudiconv/dicoms.py

```diff
@@ -62,6 +62,14 @@ def create_seqinfo(mw, series_files, series_id):
     global total_files
     total_files += len(series_files)
 
+    custom_seqinfo_data = custom_seqinfo(wrapper=mw, series_files=series_files) \
+        if custom_seqinfo else None
+    try:
+        hash(custom_seqinfo_data)
```

Comment:
@bpinsard remind me why `hashability` is the criteria instead of being serializable to string within .tsv?

Reply:
Otherwise it crashes when it tries using the seqinfo as a dict key here:

```
Traceback (most recent call last):
  File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/basile/data/src/heudiconv/heudiconv/cli/run.py", line 177, in <module>
    main()
  File "/home/basile/data/src/heudiconv/heudiconv/cli/run.py", line 24, in main
    workflow(**kwargs)
  File "/home/basile/data/src/heudiconv/heudiconv/main.py", line 327, in workflow
    study_sessions = get_study_sessions(dicom_dir_template, files,
  File "/home/basile/data/src/heudiconv/heudiconv/parser.py", line 178, in get_study_sessions
    seqinfo_dict = group_dicoms_into_seqinfos(
  File "/home/basile/data/src/heudiconv/heudiconv/dicoms.py", line 310, in group_dicoms_into_seqinfos
    seqinfos[key][seqinfo] = series_files
TypeError: unhashable type: 'dict'
```